src/client/pfs/pfs.go

```diff
@@ -1,8 +1,36 @@
 package pfs
 
-import "fmt"
+import (
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"hash"
+)
+
+var (
+	// ChunkSize is the size of file chunks when resumable upload is used
+	ChunkSize = int64(16 * 1024 * 1024) // 16 MB
+)
 
 // FullID prints repoName/CommitID
 func (c *Commit) FullID() string {
 	return fmt.Sprintf("%s/%s", c.Repo.Name, c.ID)
 }
+
+// NewHash returns a hash that PFS uses internally to compute checksums.
+func NewHash() hash.Hash {
+	return sha512.New()
```

Comment:
Why shouldn't we use CRCs here? See #2042

Reply:
The problem with switching to a new hashing algorithm is that it'd require migrations with existing deployments (since the hashes are effectively persisted), so I'm not sure if the benefits outweight the costs here.

Reply:
I didn't notice that this code was simply moved from another file. I thought this was hashing specifically for resumable uploads. Sorry for the noise.  It's an issue for a separate PR.
As a side note, CRCs are not a replacement for proper hashes since they are not 100% accurate. They accompany hashes.
CRCs don't have to be available but when they are, you are likely to speed up comparison between blocks.
What I meant is to add an optional CRC field for Block and if it's not nil compare it to the other block.